test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatAddPkgTests.cs

```diff
@@ -109,20 +122,107 @@ public class XPlatAddPkgTests
+        {
+            // Arrange
+            using (var testDirectory = TestDirectory.Create())
+            {
+                var projectPath = Path.Combine(testDirectory, project);
+                var frameworks = MSBuildStringUtility.Split(frameworkString);
+                var sources = MSBuildStringUtility.Split(sourceString);
+                File.Create(projectPath).Dispose();
+
+                var argList = new List<string>() {
+                "add",
+                packageOption,
+                package,
+                dgFileOption,
+                dgFilePath,
+                projectOption,
+                projectPath};
+
+                if (!string.IsNullOrEmpty(versionOption))
+                {
+                    argList.Add(versionOption);
+                }
+                if (!string.IsNullOrEmpty(version))
+                {
+                    argList.Add(version);
+                }
+                if (!string.IsNullOrEmpty(frameworkOption))
+                {
+                    foreach (var framework in frameworks)
+                    {
+                        argList.Add(frameworkOption);
+                        argList.Add(framework);
+                    }
+                }
+                if (!string.IsNullOrEmpty(sourceOption))
+                {
+                    foreach (var source in sources)
+                    {
+                        argList.Add(sourceOption);
+                        argList.Add(source);
+                    }
+                }
+                if (!string.IsNullOrEmpty(packageDirectoryOption))
+                {
+                    argList.Add(packageDirectoryOption);
+                    argList.Add(packageDirectory);
+                }
+                if (!string.IsNullOrEmpty(noRestoreSwitch))
+                {
+                    argList.Add(noRestoreSwitch);
+                }
+                if (!string.IsNullOrEmpty(interactiveSwitch))
+                {
+                    argList.Add(interactiveSwitch);
+                }
+                if (!string.IsNullOrEmpty(prereleaseOption))
+                {
+                    argList.Add(prereleaseOption);
+                }
+
+                var logger = new TestCommandOutputLogger();
+                var testApp = new CommandLineApplication();
+                var mockCommandRunner = new Mock<IPackageReferenceCommandRunner>();
+                mockCommandRunner
+                    .Setup(m => m.ExecuteCommand(It.IsAny<PackageReferenceArgs>(), It.IsAny<MSBuildAPIUtility>()))
+                    .ReturnsAsync(0);
+
+                testApp.Name = "dotnet nuget_test";
+                AddPackageReferenceCommand.Register(testApp,
+                    () => logger,
+                    () => mockCommandRunner.Object);
+
+                // Act
+                // Assert
+                Assert.Throws<ArgumentException>(() => testApp.Execute(argList.ToArray()));
```

Comment:
Can you verify the exception text? 
I think you might be able to get the expected string value similarly as you did in the code where you throw the exception. 

Look at InternalsVisibleTo in the project files if intellisense can't find it.